File: Stack/Evaluate_Reverse_Polish_Notation/1_BruteForce.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        // Brute‑force: repeatedly find the first operator and collapse three tokens.
        while (tokens.size() > 1) {
            size_t opIdx = 0;
            // locate first operator
            for (size_t i = 0; i < tokens.size(); ++i) {
                const string &t = tokens[i];
                if (t == "+" || t == "-" || t == "*" || t == "/") {
                    opIdx = i;
                    break;
                }
            }
            // operands are at opIdx-2 and opIdx-1
            int a = stoi(tokens[opIdx - 2]);
            int b = stoi(tokens[opIdx - 1]);
            int res = 0;
            const string &op = tokens[opIdx];
            if (op == "+") res = a + b;
            else if (op == "-") res = a - b;
            else if (op == "*") res = a * b;
            else if (op == "/") res = a / b; // truncates toward zero
            // replace the three tokens with the result string
            tokens[opIdx - 2] = to_string(res);
            // erase the two tokens that are now obsolete
            tokens.erase(tokens.begin() + opIdx - 1, tokens.begin() + opIdx + 1);
        }
        return stoi(tokens[0]);
    }
};

File: Stack/Evaluate_Reverse_Polish_Notation/1_BruteForce.java
import java.util.*;

class Solution {
    public int evalRPN(String[] tokens) {
        // Brute‑force: collapse the first operator found each iteration.
        List<String> list = new ArrayList<>(Arrays.asList(tokens));
        while (list.size() > 1) {
            int opIdx = -1;
            for (int i = 0; i < list.size(); i++) {
                String t = list.get(i);
                if (t.equals("+") || t.equals("-") || t.equals("*") || t.equals("/")) {
                    opIdx = i;
                    break;
                }
            }
            int a = Integer.parseInt(list.get(opIdx - 2));
            int b = Integer.parseInt(list.get(opIdx - 1));
            int res = 0;
            String op = list.get(opIdx);
            switch (op) {
                case "+": res = a + b; break;
                case "-": res = a - b; break;
                case "*": res = a * b; break;
                case "/": res = a / b; break; // truncates toward zero
            }
            // replace three tokens with result
            list.set(opIdx - 2, Integer.toString(res));
            // remove the two now‑redundant tokens
            list.remove(opIdx - 1);
            list.remove(opIdx - 1); // after previous removal, the next token shifts left
        }
        return Integer.parseInt(list.get(0));
    }
}
